### PR TITLE
Add binned stellar mass - BH mass plot

### PR DESCRIPTION
**Pull request by FHusko**

### CODE REVIEW

File: colibre/scripts/bh_accretion_evolution.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 from swiftpipeline.argumentparser import ScriptArgumentParser
 
 from velociraptor.observations import load_observation
+from velociraptor.observations import load_observations
```
Comment (EvgeniiChaikin) [inline colibre/scripts/bh_accretion_evolution.py:19]: This line is a duplicate of the line above
```suggestion
```

File: colibre/auto_plotter/black_holes.yml [hunk 2]
```diff
@@ -67,6 +67,39 @@ stellar_mass_black_hole_mass_50:
...
+    end:
+      value: 1e12
+      units: Solar_Mass
+  metadata:
+    title: Stellar Mass-Black Hole Mass relation (50 kpc Stellar Mass)
+    caption: SMBHM relation. The binned observational data are obtained by combining BH mass measurements for galaxies of different morphologies (Graham 2023) and passive vs. active galaxies (Terrazas et al. 2017) with dependencies on stellar mass of morphological types (Moffett et al. 2016, from the GAMA survey) and quenched fractions (Gilbank et al. 2010, SDSS).
+    section: Black Holes
+  observational_data:
+    - filename: GalaxyStellarMassBlackHoleMassBinned/Terrazas2017_binned.hdf5
+    - filename: GalaxyStellarMassBlackHoleMassBinned/Graham2023_binned.hdf5
```
Comment (EvgeniiChaikin) [inline colibre/auto_plotter/black_holes.yml:100]: I think there is a typo in the path to the observational data
```suggestion
    - filename: GalaxyStellarMassBlackHoleMass/Terrazas2017_binned.hdf5
    - filename: GalaxyStellarMassBlackHoleMass/Graham2023_binned.hdf5
```

**Review by EvgeniiChaikin — CHANGES_REQUESTED** [review]
Thank you for the nice update @FHusko!

Please address a few small comments in the code, and update the version of the `observational data` submodule. Then we can merge this pull request.

File: colibre/scripts/bh_accretion_evolution.py (hunk 1 above)
Comment (FHusko) [reply colibre/scripts/bh_accretion_evolution.py:18]: Thanks, hadn't noticed it!

**Comment by FHusko** [comment]
Thanks for the comments, @EvgeniiChaikin. Could you please remind me how I can update the reference to the observational data? I had forgotten that needs to be done, I thought it automatically points to the current master.

Unrelated to that, I am working to add very soon add high-redshift BH mass - stellar mass data, as well as BH mass functions. Would you prefer to add these as part of this PR, or a separate one?

**Comment by EvgeniiChaikin** [comment]
Let's finish this PR because it is almost ready; and for other, larger changes, create a separate PR.

Your repository's observational data points to the master branch that is 2 months old. 

To update the submodule: within your branch, go to the observational data submodule, then `checkout` from the detached state to the master branch, then do `git pull` on the master branch, and then check out to the latest commit in the master branch. Your pipeline-configs branch, which you are trying to merge here, should pick up that change in the state of the submodule. Just wrap that change into a new commit and push it here.

**Comment by FHusko** [comment]
Okay, thanks! I think it should now be pointing to the correct version of the submodule.

**Comment by EvgeniiChaikin** [comment]
Something went wrong: the submodule has not been updated fully. Its state has changed from being 2 months old to being 2 weeks old, but, for example, there was a comment yesterday to the master branch, which I cannot see in your branch.

Namely, based on your branch, I see
Screenshot from 2024-03-22 13-44-57

Screenshot from 2024-03-22 13-45-09


But the correct recent history of the observational data submodule should be https://github.com/SWIFTSIM/velociraptor-comparison-data/commits/master/

**Comment by FHusko** [comment]
I hope it is finally matching.

**Review by EvgeniiChaikin — APPROVED** [review]

**Comment by EvgeniiChaikin** [comment]
Thanks, all looks good!!

**Comment by EvgeniiChaikin** [comment]
The last remaining step before merging is to rebase to the latest version of the master branch `pipeline-configs`.

**Comment by FHusko** [comment]
When I do this (`git rebase master`) it says that it is up-to-date.

**Comment by EvgeniiChaikin** [comment]
On GitHub, I see that your branch is 3 commits behind.
Screenshot from 2024-03-24 14-30-07

I just noticed that you are merging a branch from a forked repository, instead of a branch from the original repository. This is why it may be more complicated to synchronize with the master branch of the original repository.

One problem with being out of sync, is that even though the submodule in your repo is now correct and up to date, in the file changes of this MR, I still see that the submodule is being updated, which should not be the case.

**Comment by FHusko** [comment]
Thanks for noticing! Should be in sync now :)

**Comment by EvgeniiChaikin** [comment]
Thanks! Now everything is indeed in sync. I've just merged the branch.

**Pull request merged**